src/MacroExpander.js

```diff
@@ -140,7 +140,7 @@ export default class MacroExpander implements MacroContextInterface {
                         expansion.splice(i + 1, 1); // drop first #
                     } else if (/^[1-9]$/.test(tok.text)) {
                         // replace the placeholder with the indicated argument
-                        expansion.splice(i, 2, ...args[tok.text - 1]);
+                        expansion.splice(i, 2, ...args[+tok.text - 1]);
```

Comment:
Why the `+`?

Reply:
To convert the string to a number (see second commit). When making the type of `Token`'s `text` explicitly `string` (which we thought was previously unnecessary) instead of `*`, this line through a type error (`string` and `number` couldn't be added).
I was surprised, as I expected `Token`'s `text` to have been inferred as `string` due to its constructor. Flow seems to be doing some global type inference, giving some surprising results here. Hence, we should avoid using `*` for maximal type safety.
Thanks for asking. ;)